### PR TITLE
Enhance editorial creator settings and creator attributes

### PR DESCRIPTION
**Pull request by MustafaJafar**
## Changelog Description
Follow up PR to https://github.com/ynput/ayon-traypublisher/pull/44 and https://github.com/ynput/ayon-traypublisher/pull/47

This PR is specifically for enhancing editorial creator settings
- It renames/rearranges some settings and adds description and docstrings.
image
- it also updates editorial creator attributes
image

## Testing notes:
1. Go through settings and validate if the changes enhances the UX of the settings.
2. This PR is backward compatible as it only change the settings' label, copy settings should work fine.

### CODE REVIEW

File: server/settings/editorial_creators.py [hunk 1]
```diff
@@ -39,32 +47,43 @@ class ShotRenameSubmodel(BaseSettingsModel):
...
     _layout = "compact"
     # TODO - was 'type' must be renamed in code to `parent_type`
-    parent_type: str = SettingsField(
-        "Project",
-        enum_resolver=lambda: parent_type_enum
-    )
     name: str = SettingsField(
         "",
-        title="Parent token name",
+        title="Token name",
```
Comment (iLLiCiTiT) [inline server/settings/editorial_creators.py:52]: Wasn't the `Parent token name` more descriptive? (Just asking, this is out of my scope.)

Comment (MustafaJafar) [reply server/settings/editorial_creators.py:52]: IMO, `parent_token_name` is descriptive when looking closely. 
From user perspective, I'd prefer to see the setting as whole.
Therefore, I can tell that this setting describes a folder path template where each shot will be added.
image

Comment (MustafaJafar) [reply server/settings/editorial_creators.py:52]: Maybe this is better?
```suggestion
        title="Folder token name",
```

Comment (iLLiCiTiT) [reply server/settings/editorial_creators.py:52]: Like I've wrote, out of my scope, @jakubjezek001 or @robin-ynput might have some ynput.

File: server/settings/editorial_creators.py [hunk 2]
```diff
@@ -7,20 +7,28 @@
...
-    name: str = SettingsField("", title="Tokenizer name")
-    regex: str = SettingsField("", title="Tokenizer regex")
+    name: str = SettingsField("", title="Token name")
+    regex: str = SettingsField("", title="Token regex")
 
 
 class ShotAddTasksItem(BaseSettingsModel):
     _layout = "compact"
-    name: str = SettingsField('', title="Key")
+    name: str = SettingsField('', title="Task Name")
```
Comment (robin-ynput) [inline server/settings/editorial_creators.py:16]: very very picky, just for my own OCD so it matches the others. 
```suggestion
    name: str = SettingsField('', title="Task name")
```

File: server/settings/editorial_creators.py [hunk 3]
```diff
@@ -7,20 +7,28 @@
...
 
 
 class ShotRenameSubmodel(BaseSettingsModel):
+    """Shot Rename Info
+
+    When enabled, any discovered shots will be renamed based on the `shot rename template`.
+
+    The template supports both the available 
+    [template keys](https://ayon.ynput.io/docs/admin_settings_project_anatomy#available-template-keys) 
+    and tokens defined in the `Clip Name Tokenizer`.
```
Comment (robin-ynput) [inline server/settings/editorial_creators.py:30]:
```suggestion
    and tokens defined under `Clip Name Tokenizer`.
```

File: server/settings/editorial_creators.py (hunk 1 above)
Comment (robin-ynput) [reply server/settings/editorial_creators.py:52]: Maybe `Folder token name` is the most meaningful imo. 
But to be honest, this is minimal, I think the description you added and the default examples make it pretty clear.

**Review by robin-ynput — APPROVED** [review]

**Pull request merged**